src/librustc_mir/util/elaborate_drops.rs

```diff
@@ -861,6 +837,11 @@ where
     ) -> BasicBlock {
         debug!("drop_flag_reset_block({:?},{:?})", self, mode);
 
+        if unwind.is_cleanup() {
+            // The drop flag isn't read again on the unwind path, so don't
```

Comment:
is this one of those situations where we end up not calling drop on something?

Reply:
No, in this case we've just dropped the value and will never touch it again.